core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -22,6 +22,11 @@
  */
 class PageRegular extends Frontend
 {
+	/**
+	 * @var FrontendTemplate
+	 */
+	protected $Template;
```

Comment:
This property is already defined here:

https://github.com/contao/contao/blob/f842b78a8f5a78d0dd16aacc3528e84134e18f4e/core-bundle/src/Resources/contao/library/Contao/System.php#L54

Why did you add it?

Reply:
Not in 4.9...

Reply:
It's not actually defined. The docblock is kind of wrong, as not everyone inheriting from `Contao\System` will automatically have `$this->Template`. `PageRegular` is the odd one out here, because in all other classes where the legacy framework uses `$this->Template` it is covered by `__get` and `__set` and stored in `$this->arrData`. But that's not the case here, thus it needs to be defined, otherwise PHP 8.1 will cause an error.

Reply:
> Not in 4.9...

The code snippet above shows the `4.9` branch.

Reply:
Well, I don't know. For PHP 8.1 it's not there.

Reply:
> `PageRegular` is the odd one out here, because in all other classes where the legacy framework uses `$this->Template` it is covered by `__get` and `__set` and stored in `$this->arrData`. But that's not the case here

@fritzmg I don't understand. PageRegular does not define its own getters or setters, therefore it should inherit them from the parent class.

Reply:
It's not present in any Contao version. It just does not raise an error prior to PHP 8.1.

Reply:
It's just a comment. The property needs to exist.

Reply:
What is the exact error message?

Reply:
> @fritzmg I don't understand. PageRegular does not define its own getters or setters, therefore it should inherit them from the parent class.

The parent class does not provide them either.

> What is the exact error message?

https://github.com/contao/contao/runs/4521842534

```
Creation of dynamic property Contao\PageRegular::$Template is deprecated
```

Reply:
https://wiki.php.net/rfc/deprecate_dynamic_properties

Reply:
Yes, but that is not what we are doing. We are using magic getters and setters to add objects to `$this->arrObjects`.

https://github.com/contao/contao/blob/f842b78a8f5a78d0dd16aacc3528e84134e18f4e/core-bundle/src/Resources/contao/library/Contao/System.php#L133-L149

And this should continue to work in PHP 8.2 as well. So maybe we are looking at a PHP 8.2 bug here?

Reply:
Yeah but we create a new one, so that would be `__set()`, not `__get()`.

Reply:
With `__set()` it works properly. See https://3v4l.org/akWau/rfc#vgit.master.

Reply:
Yes, we should fix this in the `System` class. If I search for `$this->Template =` in PhpStorm, there are more than 20 matches that will all trigger the same warning in PHP 8.2 otherwise.

Reply:
I don't think that's a good idea. We want them to be local to the class and not stored away in some random inheritance object.

Reply:
>  I search for $this->Template = in PhpStorm, there are more than 20 matches that will all trigger the same warning in PHP 8.2 otherwise.

That's not the case. All other classes that use `$this->Template = …` already have `__set`. It's just `PageRegular` that does not.

Reply:
PR is here: #3822